### PR TITLE
Add etcd bump version test

### PR DESCRIPTION
**Pull request by bguzman-3pillar**
Adding `etcd` version bump test.

- Add the etcd version file
- Improvements on `helper` file (change function names & comments)

To test it:
```
# Using upgrade flags -> RKE2 & K3S
go test -timeout=90m -v -tags=etcd ./entrypoint/versionbump/... -expectedValue "3.5.9, v1.26" -expectedValueUpgrade "3.5.9, v1.27" -installVersionOrCommit 000ce7f68fe2f79074d10353a20e9b4f9f3685d9 -channel testing

# Using without upgrade
go test -timeout=95m -v -tags=etcd ./entrypoint/versionbump/... -expectedValue "3.5.9, v1.27"
```

### CODE REVIEW

File: entrypoint/versionbump/etcd_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			assert.PodAssertReady(),
+			assert.PodAssertStatus())
+	})
+
+	It("Verifies bump version on product for etcd", func() {
+		var cmd string
+		if cfg.Product == "rke2" {
+			cmd = "sudo /var/lib/rancher/rke2/bin/crictl -r unix:///run/k3s/containerd/containerd.sock images | grep etcd ," +
+				" rke2 -v"
+		} else {
```
Comment (fmoral2) [inline entrypoint/versionbump/etcd_test.go:39]: u dont need this else

File: pkg/template/versiontemplate.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ func VersionTemplate(test VersionTestTemplate) {
 		}
 	}
 
-	err := checkVersion(test)
+	err := executeTestCombination(test)
```
Comment (fmoral2) [inline pkg/template/versiontemplate.go:27]: every call of this func is refactored right ?

File: pkg/template/helper.go [hunk 3]
```diff
@@ -9,23 +9,28 @@ import (
...
 	if err != nil {
 		return err
 	}
 
+	updateExpectedValue(template)
+
+	return nil
+}
+
+// updateExpectedValue updates the expected values
```
Comment (fmoral2) [inline pkg/template/helper.go:24]: i would just improve a little the comment here, like saying that it updates getting the values from flag ExpectedValueUpgrade

File: pkg/template/processor.go [hunk 4]
```diff
@@ -38,12 +39,21 @@ func processCmds(resultChan chan error, wg *sync.WaitGroup, ip string, cmds []st
...
+		for _, testMap := range testCombination.Run {
+
+			cmds := strings.Split(testMap.Cmd, ",")
+			expectedValues := strings.Split(testMap.ExpectedValue, ",")
+
+			if strings.Contains(testMap.Cmd, "etcd") {
+				for _, serverIP := range cluster.ServerIPs {
+					processCmds(resultChan, wg, serverIP, cmds, expectedValues)
+				}
+			} else {
```
Comment (fmoral2) [inline pkg/template/processor.go:53]: u dont need this else as well.

File: entrypoint/versionbump/etcd_test.go (hunk 1 above)
Comment (bguzman-3pillar) [reply entrypoint/versionbump/etcd_test.go]: Fixed

File: pkg/template/versiontemplate.go (hunk 2 above)
Comment (bguzman-3pillar) [reply pkg/template/versiontemplate.go:27]: Fixed

File: pkg/template/helper.go [hunk 5]
```diff
@@ -9,23 +9,28 @@ import (
...
 	}
-
-	return nil
 }
 
-// checkVersion checks the version of RKE2 by calling processTestCombination
-func checkVersion(v VersionTestTemplate) error {
+// executeTestCombination get a template and pass it to `processTestCombination` to execute test combination on group of IPs
+func executeTestCombination(v VersionTestTemplate) error {
 	ips, err := getIPs()
```
[the remark this answers is not in the document]
Comment (bguzman-3pillar) [reply pkg/template/helper.go]: Fixed

File: pkg/template/processor.go (hunk 4 above)
Comment (bguzman-3pillar) [reply pkg/template/processor.go]: Fixed

File: entrypoint/versionbump/etcd_test.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+			cmd = "sudo /var/lib/rancher/rke2/bin/crictl -r unix:///run/k3s/containerd/containerd.sock images | grep etcd ," +
+				" rke2 -v"
+		}
+
+		template.VersionTemplate(template.VersionTestTemplate{
+			TestCombination: &template.RunCmd{
+				Run: []template.TestMap{
+					{
+						Cmd:                  cmd,
+						ExpectedValue:        template.TestMapTemplate.ExpectedValue,
```
Comment (ShylajaDevadiga) [inline entrypoint/versionbump/etcd_test.go:46]: Can we change this to currentValue and other to upgradedValue? 
For non-upgrade scenarios current and upgradedValue would be same. If upgradedValue is not passed currentValue=upgradedValue.

File: pkg/template/processor.go [hunk 7]
```diff
@@ -39,21 +38,40 @@ func processCmds(resultChan chan error, wg *sync.WaitGroup, ip string, cmds []st
...
 
 func processTestCombination(resultChan chan error, wg *sync.WaitGroup, ips []string, testCombination RunCmd) {
-	cluster := factory.GetCluster(GinkgoT())
 	if testCombination.Run != nil {
 		for _, testMap := range testCombination.Run {
 
 			cmds := strings.Split(testMap.Cmd, ",")
 			expectedValues := strings.Split(testMap.ExpectedValue, ",")
 
+			ipsToUse := ips
```
Comment (fmoral2) [inline pkg/template/processor.go:47]: why are u doing this assignment ?

File: pkg/template/processor.go [hunk 8]
```diff
@@ -39,10 +39,38 @@ func processCmds(resultChan chan error, wg *sync.WaitGroup, ip string, cmds []st
 
 func processTestCombination(resultChan chan error, wg *sync.WaitGroup, ips []string, testCombination RunCmd) {
 	if testCombination.Run != nil {
-		for _, ip := range ips {
-			for _, testMap := range testCombination.Run {
-				cmds := strings.Split(testMap.Cmd, ",")
-				expectedValues := strings.Split(testMap.ExpectedValue, ",")
+		for _, testMap := range testCombination.Run {
+
```
Comment (fmoral2) [inline pkg/template/processor.go:43]: extra blank line

File: pkg/template/processor.go [hunk 9]
```diff
@@ -39,10 +39,38 @@ func processCmds(resultChan chan error, wg *sync.WaitGroup, ip string, cmds []st
...
+			cmds := strings.Split(testMap.Cmd, ",")
+			expectedValues := strings.Split(testMap.ExpectedValue, ",")
+
+			ipsToUse := ips
+
+			if strings.Contains(testMap.Cmd, "etcd") {
+
```
Comment (fmoral2) [inline pkg/template/processor.go:50]: extra blank line

Comment (bguzman-3pillar) [reply pkg/template/processor.go]: Fixed

File: pkg/template/processor.go (hunk 8 above)
Comment (bguzman-3pillar) [reply pkg/template/processor.go]: Fixed

**Review by fmoral2 — APPROVED** [review]

**Review by rancher-max — APPROVED** [review]

**Review by mdrahman-suse — APPROVED** [review]
LGTM

**Pull request merged**